src/include/txn_inline.h

```diff
@@ -1088,59 +1082,66 @@ __wt_txn_read(
          */
         have_stop_tw = WT_TIME_WINDOW_HAS_STOP(&cbt->upd_value->tw);
 
-        /* Check the ondisk value. */
-        ret = __wt_value_return_buf(cbt, cbt->ref, &cbt->upd_value->buf, &tw
-#ifdef HAVE_DIAGNOSTIC
-          ,
-          &is_ovfl_rm
-#endif
-        );
-        WT_RET(ret);
-
-        /*
-         * If the stop time point is set, that means that there is a tombstone at that time. If it
-         * is not prepared and it is visible to our txn it means we've just spotted a tombstone and
-         * should return "not found", except scanning the history store during rollback to stable
-         * and when we are told to ignore non-globally visible tombstones.
-         */
-        if (!have_stop_tw && __wt_txn_tw_stop_visible(session, &tw) &&
-          !F_ISSET(&cbt->iface, WT_CURSTD_IGNORE_TOMBSTONE)) {
-            cbt->upd_value->buf.data = NULL;
-            cbt->upd_value->buf.size = 0;
-            cbt->upd_value->tw.durable_stop_ts = tw.durable_stop_ts;
-            cbt->upd_value->tw.stop_ts = tw.stop_ts;
-            cbt->upd_value->tw.stop_txn = tw.stop_txn;
-            cbt->upd_value->tw.prepare = tw.prepare;
-            cbt->upd_value->type = WT_UPDATE_TOMBSTONE;
-            return (0);
-        }
-
-        /* Store the stop time pair of the history store record that is returning. */
-        if (!have_stop_tw && WT_TIME_WINDOW_HAS_STOP(&tw) && WT_IS_HS(session->dhandle)) {
-            cbt->upd_value->tw.durable_stop_ts = tw.durable_stop_ts;
-            cbt->upd_value->tw.stop_ts = tw.stop_ts;
-            cbt->upd_value->tw.stop_txn = tw.stop_txn;
-            cbt->upd_value->tw.prepare = tw.prepare;
-        }
+        if (read_onpage) {
```

Comment:
By adding timing stress here, is it possible to generate the scenario using a python test?

Reply:
Adding a timing stress is a good idea. Hopefully that can help reproduce this in the test format reliably and there is no need for the python test.

Reply:
I added the timing stress and it is producing easily.

Reply:
@agorrod With the timing stress, the issue is consistently reproducing with the test format config I added.

Reply:
Great - that's good enough for me.